benchmark/README.md

```diff
@@ -57,3 +57,50 @@ ok  	github.com/wangkuiyi/gotorch	33.026s
 The throughput: 21426.107422 samples/sec
        14.37 real        83.01 user         0.64 sys
 ```
+
+## Observations
+
+We have the following observations on the throughput:
+
+1. The throughput of the GoTorch version is about **2 times** of the PyTorch
+   version.
+1. The throughput of the C++ version is about **2 times** of the GoTorch version.
+
+We have the following observations on the total run time:
+
+1. The GoTorch version takes **34 seconds**,
+1. The PyTorch version takes about **71 seconds**.
+1. The C++ version that calls libtorch directly takes only **14 seconds**.
+
+We have the following observation on the ease of programming.
+
+1. The GoTorch version has almost the same number of lines of code as the PyTorch
+   version.
+1. The C++ version is lengthy and challenging to read or write.  Thanks to our 
+   friends from Facebook PyTorch team, without their help, we cannot write the
+   C++ version and well-benchmark it.
+
+## Analysis
+
+**Why is the GoTorch version runs that much faster than the PyTorch version?**
+
+Diving into details, we found that the libtorch installed from the
+`pip install` command and used by the PyTorch version doesn't allow MKL and
+OpenMP to use all CPU cores.
+
+We verified this fact in the additional
+[trial](https://github.com/wangkuiyi/gotorch/pull/105#issuecomment-672336636), 
+where we force both the GoTorch and PyTorch version to use only one CPU core,
+they have similar throughput -- in particular, the GoTorch version is 22\%
+faster than the PyTorch version.
+
+**Why is the C++ version runs that much faster than the Go version?**
+
+The reasons include:
+
+1. A C++ function call usually takes a few ns; however, it takes about 15μm for
+   a Go function to call a C/C++ function.
+1. The C++ program frees tensors out-of-scope using reference count maintained
```

Comment:
This is probably not the reason. the `GC`-caused overhead include:
1. Mark-and-sweep the unreachable `Go` variables
    - In a `gotorch` program, there're only a few `Go` variables.
2. Call C++ `new` upon cgo function call and `delete` upon the end of an iteration.
    - Only the `delete`s and `new`s are additional overhead in each iteration.

It seems that GC does not cause much overhead

Reply:
Agreed. We should dig in for more plausible reasons,.